core/arch/arm/mm/core_mmu.c

```diff
@@ -988,6 +988,78 @@ static void set_pg_region(struct core_mmu_table_info *dir_info,
 	}
 }
 
+TEE_Result core_mmu_map_pages(vaddr_t start, paddr_t *pages, size_t num_pages,
+			      enum teecore_memtypes memtype)
+{
+	TEE_Result ret;
+	struct core_mmu_table_info tbl_info;
+	unsigned int idx;
+	uint32_t old_attr;
+	vaddr_t vaddr = start;
+	size_t i;
+
+	if (vaddr & SMALL_PAGE_MASK)
+		return TEE_ERROR_BAD_PARAMETERS;
```

Comment:
Looking back at the whole and how it is used later, I understand that these check around arguments can be useless. So using only `assert()` should be fine for one to build a safe core.

Below call to `if (!core_mmu_is_vaspace(mm)) panic();` checks the requested vmem is ok, but does not check coherency between `vaddr` and the target mapping attributes (`memtype`).
Actually, the only constraint I would really assert is mapping executable memory.
Maybe this is sufficient at entry of `core_mmu_map_pages()`:
```c
	assert(vspace_is_dynmap(vaddr, num_pages * SMALL_PAGE_SIZE, memtype));
```
with
```c
static bool __maybe_unused vspace_is_dynmap(vaddr_t vaddr, size_t size,
					    enum teecore_memtypes memtype)
{
	struct tee_mmap_region *mm = find_map_by_va((void *)vaddr);

	return (memtype != MEM_AREA_TEE_RAM &&
		va_is_in_map(mm, vstart + size) &&
		(mm->type == MEM_AREA_RES_VASPACE ||
			mm->type == MEM_AREA_SHM_VASPACE) &&
		(mm->type != MEM_AREA_SHM_VASPACE ||
			memtype == MEM_AREA_SHM_VASPACE));
}
```